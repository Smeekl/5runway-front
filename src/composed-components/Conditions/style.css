.conditions {
  padding: 10px 10px 10px 10px;
  border: 2px solid var(--color-primary-blue);
  border-radius: 10px;
  width: 700px;
  height: 650px;
}

.conditions__header {
  display: flex;
  justify-content: space-around;
}
.condition__title {
  font-size: 25px;
  font-weight: bold;
  color: var(--color-secondary);
}

.condition__items {
  padding: 5px 0px 5px 0px;
  margin: 0;
  font-size: 18px;
  /* font-weight: bold; */
  color: var(--color-secondary);
}

.condition__check {
  padding-top: 30px;
  display: flex;
  justify-content: space-evenly;
}

.container {
  display: block;
  position: relative;
  padding-left: 35px;
  margin-bottom: 12px;
  cursor: pointer;
  font-size: 22px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}
.container .check-highload {
  position: absolute;
  opacity: 0;
}
.highload2 {
  position: absolute;
  top: 0;
  left: 0;
  height: 25px;
  width: 25px;
  border: 2px solid var(--color-primary-blue);
}
.container:hover .check-highload ~ .highload2 {
  background-color: var(--color-primary-blue);
  border: solid 1px var(--color-primary-blue);
}
.container .check-highload:checked ~ .highload2 {
  background-color: var(--color-primary-blue);
}
.highload2:after {
  content: '';
  position: absolute;
  display: none;
}
.container .check-highload:checked ~ .highload2:after {
  display: block;
}
.container .highload2:after {
  left: 8px;
  top: 4px;
  width: 8px;
  height: 12px;
  border: solid var(--color-secodary-white);
  border-width: 0 4px 4px 0;
  transform: rotate(45deg);
}

.check-cursor {
  cursor: pointer;
}
