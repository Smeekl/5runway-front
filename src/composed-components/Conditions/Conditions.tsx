import { useState } from 'react';
import { ReactComponent as Logo } from '@svgIcons/iconLogo.svg';
import './style.css';

export const Conditions = () => {
  const [isChecked, setIsChecked] = useState(false);

  return (
    <div className="conditions">
      <div className="conditions__header">
        <span>
          <Logo />
        </span>
        <p className="condition__title">Terms and Conditions</p>
      </div>
      <div>
        <p className="condition__items">
          Generally, Terms & Conditions are not a legal requirement, though, in
          many instances its in your best interest to have one in place. This is
          because Terms & Conditions are necessary for governing the contractual
          relationship between you and your users. They are essential for
          protecting your content from a copyright perspective as well as
          protecting you from potential liabilities.
        </p>
        <p className="condition__items">
          Youll likely need a Terms & Conditions document if any of the
          following apply to you:
        </p>
        <p className="condition__items">
          1. You have different user levels (eg. registered vs non-registered);
        </p>
        <p className="condition__items">
          2. You want to set the rules for user behavior (including comments)
          and state grounds for termination of accounts;
        </p>
        <p className="condition__items">
          3. Your users are allowed to upload content;
        </p>

        <p className="condition__items">
          4. You participate in some kind of commerce, including affiliate
          programs;
        </p>
        <p className="condition__items">
          5. Youd like to protect your website/app and its content by stating
          how it can be used.
        </p>
      </div>
      <div className="condition__check">
        <label className="container">
          <input
            className="check-highload"
            type="checkbox"
            checked={isChecked}
            onChange={() => {
              setIsChecked(!isChecked);
            }}
          />
          <span className="highload2"></span>
        </label>

        <p
          className="condition__items check-cursor"
          onClick={() => {
            setIsChecked(!isChecked);
          }}
        >
          I have read this it and agree to the terms conditions
        </p>
      </div>
    </div>
  );
};
