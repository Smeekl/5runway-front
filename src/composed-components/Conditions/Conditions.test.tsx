import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Conditions } from './Conditions';

describe('Conditions UI tests', () => {
  test('should render correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Conditions />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test('should be in the document', () => {
    const { container } = render(
      <MemoryRouter>
        <Conditions />
      </MemoryRouter>
    );
    expect(container).toBeInTheDocument();
  });
});
