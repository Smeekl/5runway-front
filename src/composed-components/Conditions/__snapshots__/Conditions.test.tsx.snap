// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main UI tests should render correctly 1`] = `
<div>
  <div
    class="conditions"
  >
    <div
      class="conditions__header"
    >
      <span>
        <reactcomponent />
      </span>
      <p
        class="condition__title"
      >
        Terms and Conditions
      </p>
    </div>
    <div>
      <p
        class="condition__items"
      >
        Generally, Terms & Conditions are not a legal requirement, though, in many instances its in your best interest to have one in place. This is because Terms & Conditions are necessary for governing the contractual relationship between you and your users. They are essential for protecting your content from a copyright perspective as well as protecting you from potential liabilities.
      </p>
      <p
        class="condition__items"
      >
        Youll likely need a Terms & Conditions document if any of the following apply to you:
      </p>
      <p
        class="condition__items"
      >
        1. You have different user levels (eg. registered vs non-registered);
      </p>
      <p
        class="condition__items"
      >
        2. You want to set the rules for user behavior (including comments) and state grounds for termination of accounts;
      </p>
      <p
        class="condition__items"
      >
        3. Your users are allowed to upload content;
      </p>
      <p
        class="condition__items"
      >
        4. You participate in some kind of commerce, including affiliate programs;
      </p>
      <p
        class="condition__items"
      >
        5. Youd like to protect your website/app and its content by stating how it can be used.
      </p>
    </div>
    <div
      class="condition__check"
    >
      <label
        class="container"
      >
        <input
          class="check-highload"
          type="checkbox"
        />
        <span
          class="highload2"
        />
      </label>
      <p
        class="condition__items check-cursor"
      >
        I have read this it and agree to the terms conditions
      </p>
    </div>
  </div>
</div>
`;
